tests/run/i11050.scala

```diff
@@ -0,0 +1,68 @@
+import scala.compiletime.*
+import scala.deriving.*
+
+sealed trait TreeValue
+
+sealed trait SubLevel extends TreeValue
+
+case class Leaf1(value: String) extends TreeValue
+case class Leaf2(value: Int)    extends SubLevel
+case class Leaf3(value: Char)   extends TreeValue
+
+object Test:
+  val m                 = summon[Mirror.SumOf[TreeValue]]
+  given Show[TreeValue] = Show.derived[TreeValue]
+
+  def main(args: Array[String]) =
+    val leaf1 = Leaf1("1")
+    val leaf2 = Leaf2(2)
+    val leaf3 = Leaf3('3')
+
+    assertEq(List(leaf1, leaf2, leaf3).map(m.ordinal), List(0, 1, 2))
+    assertShow(leaf1, "Leaf1(value = \"1\")")
```

Comment:
Aren’t you summuning an instance for the case class `Leaf1` here rather than for the trait `TreeValue`? What happens with this?

```suggestion
    assertShow[TreeValue](leaf1, "Leaf1(value = \"1\")")
```